tests/test_atoms_state.py

```diff
@@ -0,0 +1,350 @@
+        assert resolved_degeneracy == degeneracy
+
+    def test_normalize(self):
+        """
+        Test the normalization of the `OrbitalsState`. Inputs are defined as the quantities of the `OrbitalsState` section.
+        """
+        orbital_state = OrbitalsState(n_quantum_number=2)
+        self.add_state(orbital_state, 2, -2, None, None, None)
+        orbital_state.normalize(None, logger)
+        assert orbital_state.n_quantum_number == 2
+        assert orbital_state.l_quantum_number == 2
+        assert orbital_state.l_quantum_symbol == 'd'
+        assert orbital_state.ml_quantum_number == -2
+        assert orbital_state.ml_quantum_symbol == 'xy'
+        assert orbital_state.degeneracy == 2
+
+
+class TestCoreHole:
+    """
+    Test the `CoreHole` class defined in atoms_state.py.
+    """
+
+    @pytest.mark.parametrize(
+        'orbital_ref, degeneracy, n_excited_electrons, occupation',
+        [
+            (OrbitalsState(l_quantum_number=1), 6, 0.5, 5.5),
+            (OrbitalsState(l_quantum_number=1, ml_quantum_number=-1), 2, 0.5, 1.5),
+            (None, None, 0.5, None),
+        ],
+    )
+    def test_occupation(self, orbital_ref, degeneracy, n_excited_electrons, occupation):
+        """
+        Test the occupation of a core hole for a given set of orbital reference and degeneracy.
+        """
+        core_hole = CoreHole()
+        core_hole.orbital_ref = orbital_ref
+        if orbital_ref is not None:
+            assert orbital_ref.resolve_degeneracy() == degeneracy
+        core_hole.n_excited_electrons = n_excited_electrons
+        resolved_occupation = core_hole.resolve_occupation(logger)
+        if resolved_occupation is not None:
+            assert np.isclose(resolved_occupation, occupation)
+        else:
+            assert resolved_occupation == occupation
+
+    @pytest.mark.parametrize(
+        'orbital_ref, n_excited_electrons, dscf_state, results',
+        [
+            (OrbitalsState(l_quantum_number=1), -0.5, None, (-0.5, None, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, None, (0.5, None, 5.5)),
+            (
+                OrbitalsState(l_quantum_number=1, ml_quantum_number=-1),
+                0.5,
+                None,
+                (0.5, None, 1.5),
+            ),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'initial', (None, 1, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'final', (0.5, None, 5.5)),
+            (None, 0.5, None, (0.5, None, None)),
+        ],
+    )
+    def test_normalize(self, orbital_ref, n_excited_electrons, dscf_state, results):
+        """
+        Test the normalization of the `CoreHole`. Inputs are defined as the quantities of the `CoreHole` section.
+        """
+        core_hole = CoreHole()
+        core_hole.orbital_ref = orbital_ref
+        core_hole.n_excited_electrons = n_excited_electrons
+        core_hole.dscf_state = dscf_state
```

Comment:
Why not perform all the setting at instantiation?

Reply:
True, I moved it